### PR TITLE
super-tiny-icons: init at 2022-11-07

### PR DESCRIPTION
**Pull request by h7x4**
###### Description of changes

Packaged https://github.com/edent/SuperTinyIcons as a standalone icon package.

It is useful for all sorts of modules that uses icons, e.g [`services.jupyter.kernels.<name>.logo64`](https://search.nixos.org/options?channel=22.05&show=services.jupyter.kernels.%3Cname%3E.logo64&from=0&size=50&sort=relevance&type=packages&query=services.jupyter.)

This depends on #199767, because I added my maintainer entry there.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (SuperSandro2000) [inline pkgs/data/icons/super-tiny-icons/default.nix:15]:
```suggestion
```
Doing this in packages is forbidden. Please disable other phases if they cause issues.

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (SuperSandro2000) [inline pkgs/data/icons/super-tiny-icons/default.nix:10]:
```suggestion

  src = fetchFromGitHub {
    owner = "edent";
    repo = "SuperTinyIcons";
    rev = "b4d5a3be04c99ec0a309ac9e0d0b21207c237c7d";
```

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (SuperSandro2000) [inline pkgs/data/icons/super-tiny-icons/default.nix:4]:
```suggestion
{ pkgs, lib, stdenvNoCC, fetchFromGitHub }:

stdenvNoCC.mkDerivation rec {
  pname = "super-tiny-icons";
  version = "unstable-2022-11-07";
```

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (SuperSandro2000) [inline pkgs/data/icons/super-tiny-icons/default.nix:31]: We normally do not copy all this stuff because NixOS has no easy way to browse them all.

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (SuperSandro2000) [inline pkgs/data/icons/super-tiny-icons/default.nix:14]:
```suggestion
```
I don't think we need an extra output for a couple of md files.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please carry the maintainer change over from the other PR https://github.com/NixOS/nixpkgs/pull/199767/commits/50b98fff1a6ffe8341e627bea48ab4de6c04d0bd#diff-c1fcec3afa1da96245de1d7954ca721c979a5f25aee0e8b59d59d3d21e31f8fa

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (h7x4) [reply pkgs/data/icons/super-tiny-icons/default.nix]: That is good to know :+1:

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (pbsds) [inline pkgs/data/icons/super-tiny-icons/default.nix:19]:
```suggestion
      cp -r "$icondir" $out/share/icons/SuperTinyIcons/
```

File: pkgs/data/icons/super-tiny-icons/default.nix
Comment (pbsds) [inline pkgs/data/icons/super-tiny-icons/default.nix:20]: In general I dislike having bash split the output of find.
Perhaps something like this?
```suggestion
    find $src/images -type d -exec cp -r {} $out/share/icons/SuperTinyIcons/ \;
```

**Comment by pbsds** [comment]
LGTM after a squash!

**Review by pbsds — APPROVED** [review]

**Pull request merged**
